embedchain/vectordb/pineconedb.py

```diff
@@ -0,0 +1,180 @@
+import copy
+import os
+from typing import Dict, List, Optional
+
+try:
+    import pinecone
+except ImportError:
+    raise ImportError(
+        "Pinecone requires extra dependencies. Install with `pip install --upgrade embedchain[pinecone]`"
+    ) from None
+
+from embedchain.config.vectordb.pinecone import PineconeDbConfig
+from embedchain.helper.json_serializable import register_deserializable
+from embedchain.vectordb.base import BaseVectorDB
+
+
+@register_deserializable
+class PineconeDb(BaseVectorDB):
+    BATCH_SIZE = 100
+
+    """
+    Pinecone as vector database
+    """
+
+    def __init__(
+        self,
+        config: Optional[PineconeDbConfig] = None,
+    ):
+        """Pinecone as vector database.
+
+        :param config: Pinecone database config, defaults to None
+        :type config: PineconeDbConfig, optional
+        :raises ValueError: No config provided
+        """
+        if config is None:
+            self.config = PineconeDbConfig()
+        else:
+            if not isinstance(config, PineconeDbConfig):
+                raise TypeError(
+                    "config is not a `PineconeDbConfig` instance. "
+                    "Please make sure the type is right and that you are passing an instance."
+                )
+            self.config = config
+        self.client = self._setup_pinecone_index()
+        # Call parent init here because embedder is needed
+        super().__init__(config=self.config)
+
+    def _initialize(self):
+        """
+        This method is needed because `embedder` attribute needs to be set externally before it can be initialized.
+        """
+        if not self.embedder:
+            raise ValueError("Embedder not set. Please set an embedder with `set_embedder` before initialization.")
+
+    # Loads the Pinecone index or creates it if not present.
+    def _setup_pinecone_index(self):
+        pinecone.init(
+            api_key=os.environ.get("PINECONE_API_KEY"),
+            environment=os.environ.get("PINECONE_ENV"),
+        )
```

Comment:
Can we please error out with proper error message if the env variables are missing? Currently, it errors out without a proper error messaage.

Reply:
Pinecone throws `API key is missing or invalid for the environment "us-west1-gcp". Check that the correct environment is specified.` which is also bubbled up in the app. Thats descriptive enough for users to understand what went wrong?